### PR TITLE
Blackbox trivial test

### PR DESCRIPTION
**Pull request by jparisu**
Merge after #47

### CODE REVIEW

File: include/ddsrouter/communication/Track.hpp [hunk 1]
```diff
@@ -105,9 +105,9 @@ class Track
     /**
      * Whether this Track is enabled
      *
-     * Thread safe
+     * Not Thread safe, call
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/communication/Track.hpp:108]: Complete comment

File: src/cpp/core/DDSRouter.cpp [hunk 2]
```diff
@@ -92,7 +100,13 @@ void DDSRouter::init_participants_()
             discovery_database_);
 
         // create_participant should throw an exception in fail, never return nullptr
-        assert(nullptr != new_participant);
+        if (!new_participant || !new_participant->id().is_valid() ||
+            new_participant->type() == ParticipantType::PARTICIPANT_TYPE_INVALID)
+        {
+            // Fail creating participant
+            throw InitializationException(utils::Formatter()
+                << "Fail creating Participant " << participant_config.id());
```
Comment (juanlofer-eprosima) [inline src/cpp/core/DDSRouter.cpp:108]:
```suggestion
            // Failed to create participant
            throw InitializationException(utils::Formatter()
                << "Failed to create Participant " << participant_config.id());
```

File: src/cpp/participant/implementations/auxiliar/DummyParticipant.cpp [hunk 3]
```diff
@@ -24,14 +24,26 @@
...
+}
+
+DummyParticipant::~DummyParticipant()
+{
+    std::unique_lock<std::mutex> lock(static_mutex_);
+
+    writers_.clear();
+    readers_.clear();
+
+    // participants_[id()] = nullptr;
```
Comment (juanlofer-eprosima) [inline src/cpp/participant/implementations/auxiliar/DummyParticipant.cpp:45]:
```suggestion
    writers_.clear();
    readers_.clear();

    std::unique_lock<std::mutex> lock(static_mutex_);
```

File: src/cpp/types/RawConfiguration.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// limitations under the License.
+
+/**
+ * @file RawConfiguration.cpp
+ *
+ */
+
+#include <ddsrouter/types/RawConfiguration.hpp>
+#include <ddsrouter/types/utils.hpp>
+#include <ddsrouter/exceptions/ConfigurationException.hpp>
```
Comment (juanlofer-eprosima) [inline src/cpp/types/RawConfiguration.cpp:22]: sort

File: test/blackbox/ddsrouter_core/TrivialTest.cpp [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+#include <algorithm>
+#include <iostream>
+
+#include <gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <ddsrouter/core/DDSRouter.hpp>
+#include <ddsrouter/types/RawConfiguration.hpp>
+#include <ddsrouter/types/utils.hpp>
+#include <ddsrouter/participant/implementations/auxiliar/DummyParticipant.hpp>
```
Comment (juanlofer-eprosima) [inline test/blackbox/ddsrouter_core/TrivialTest.cpp:24]: sort

File: test/blackbox/ddsrouter_core/TrivialTest.cpp [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+using namespace eprosima::ddsrouter;
+
+Guid random_guid(uint16_t seed=0)
+{
+    Guid guid;
+    guid.entityId.value[3] = seed;
+    guid.guidPrefix.value[0] = 0x01;
+    guid.guidPrefix.value[0] = 0x0f;
```
Comment (juanlofer-eprosima) [inline test/blackbox/ddsrouter_core/TrivialTest.cpp:33]: Set value twice?

File: test/blackbox/ddsrouter_core/TrivialTest.cpp [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    return guid;
+}
+
+Payload random_payload(uint16_t seed=0)
+{
+    Payload payload;
+    payload.length = sizeof(uint16_t);
+    payload.reserve(payload.length);
+    //  = static_cast<eprosima::fastrtps::rtps::octet*>(new uint16_t(seed));
```
Comment (juanlofer-eprosima) [inline test/blackbox/ddsrouter_core/TrivialTest.cpp:42]:
```suggestion
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Please fix uncrustify test

File: include/ddsrouter/communication/Track.hpp (hunk 1 above)
Comment (jparisu) [reply include/ddsrouter/communication/Track.hpp:108]: :+1:

File: src/cpp/core/DDSRouter.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/core/DDSRouter.cpp:108]: :+1:

File: src/cpp/participant/implementations/auxiliar/DummyParticipant.cpp (hunk 3 above)
Comment (jparisu) [reply src/cpp/participant/implementations/auxiliar/DummyParticipant.cpp:45]: Why? If the mutex is not taken, a new writer could be created before the destruction has finished.

File: src/cpp/types/RawConfiguration.cpp (hunk 4 above)
Comment (jparisu) [reply src/cpp/types/RawConfiguration.cpp:22]: :+1:

File: test/blackbox/ddsrouter_core/TrivialTest.cpp (hunk 5 above)
Comment (jparisu) [reply test/blackbox/ddsrouter_core/TrivialTest.cpp:24]: :+1:

File: test/blackbox/ddsrouter_core/TrivialTest.cpp (hunk 6 above)
Comment (jparisu) [reply test/blackbox/ddsrouter_core/TrivialTest.cpp:33]: Typo :+1:

File: test/blackbox/ddsrouter_core/TrivialTest.cpp (hunk 7 above)
Comment (jparisu) [reply test/blackbox/ddsrouter_core/TrivialTest.cpp:42]: :+1:

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**